tracee-rules/signatures/golang/kubernetes_api_connection.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	tracee "github.com/aquasecurity/tracee/tracee-ebpf/external"
+	"github.com/aquasecurity/tracee/tracee-rules/signatures/helpers"
+	"github.com/aquasecurity/tracee/tracee-rules/types"
+)
+
+type K8sApiConnection struct {
+	cb                    types.SignatureHandler
+	apiAddressContainerId map[string]string
+}
+
+func (sig *K8sApiConnection) Init(cb types.SignatureHandler) error {
+	sig.cb = cb
+	sig.apiAddressContainerId = make(map[string]string)
+
+	return nil
+}
+
+func (sig *K8sApiConnection) GetMetadata() (types.SignatureMetadata, error) {
+	return types.SignatureMetadata{
+		ID:          "TRC-15",
+		Version:     "0.1.0",
+		Name:        "Kubernetes API server connection detected",
+		Description: "A connection to the kubernetes API server was detected. The K8S API server is the brain of your K8S cluster, adversaries may try and communicate with the K8S API server to gather information/credentials, or even run more containers and laterally expand their grip on your systems.",
+		Tags:        []string{"container"},
+		Properties: map[string]interface{}{
+			"Severity":     1,
+			"MITRE ATT&CK": "Discovery: Cloud Service Discovery",
+		},
+	}, nil
+}
+
+func (sig *K8sApiConnection) GetSelectedEvents() ([]types.SignatureEventSelector, error) {
+	return []types.SignatureEventSelector{
+		{Source: "tracee", Name: "execve", Origin: "container"},
+		{Source: "tracee", Name: "security_socket_connect", Origin: "container"},
+	}, nil
+}
+
+func (sig *K8sApiConnection) OnEvent(e types.Event) error {
+	eventObj, ok := e.(tracee.Event)
+	if !ok {
+		return fmt.Errorf("invalid event")
+	}
+
+	containerID := eventObj.ContainerID
+	if containerID == "" {
+		return nil
+	}
+
+	switch eventObj.EventName {
+
+	case "execve":
+		envpArg, err := helpers.GetTraceeArgumentByName(eventObj, "envp")
+		if err != nil {
+			return nil
+		}
+		envs := envpArg.Value.([]string)
+
+		apiIPAddress := getApiAddressFromEnvs(envs)
+		if apiIPAddress != "" {
```

Comment:
you can move this earlier to save a few cycles

Reply:
you mean the assertion? or the call to `getApiAddressFromEnvs`?